Use the VF2 layout algorithm to optimize for two-qubit gate count

Misty-W opened the pull request:
Fixes #85.

jordandsullivan commented:
@Misty-W can you add the improvement factors here so they're documented? :)

Misty-W commented:
> @Misty-W can you add the improvement factors here so they're documented? :)

![image](https://github.com/user-attachments/assets/2b5b500b-f21a-4fb4-8cfa-4e86aa35b357)

![image](https://github.com/user-attachments/assets/cd577aae-411a-4e41-bd17-853607223f74)

Misty-W commented:
Should I keep the results file or delete it? If I understand correctly we don't keep locally run benchmarks on `main`.

jordandsullivan commented:
Correct, we don't want to save local results files to main. You could
delete it or move it to test_data

Best,
Jordan Sullivan

On Tue, Jan 14, 2025, 11:10 AM Misty Wahl ***@***.***> wrote:

> Should I keep the results file or delete it? If I understand correctly we
> don't keep locally run benchmarks on main.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/unitaryfund/ucc/pull/169#issuecomment-2590905563>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ADYYB56TZDZ5QHEAIWVOSCL2KVOMBAVCNFSM6AAAAABVFMBXPWVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDKOJQHEYDKNJWGM>
> .
> You are receiving this because you commented.Message ID:
> ***@***.***>
>

jordandsullivan commented on ucc/transpiler_passes/layout/vf2_post_layout.py:
Have you made any changes to the Qiskit version of this file, or is it the original intact? If we've modified it, we just need to add a notice to the top saying so.

Misty-W replied:
I modified the file, so I added a notice to that effect.

jordandsullivan reviewed (commented):
Where do the tests or benchmarks live for the new mapping passes?

Misty-W commented:
>  Where do the tests or benchmarks live for the new mapping passes?
I added a simple test from Qiskit's `test_vf2_post_layout`. Since I just modified the layout scoring function, I don't think we need to run everything that's in the (long!) original file.

I also added a function for generating the layouts in PR #142, but it's not part of the automated benchmarking pipeline yet. I'm leaning toward adding that in a separate issue/PR, following our pattern for other benchmarks.

jordandsullivan reviewed (approved):
Great thanks Misty!
